web/client/api/CSW.js

```diff
@@ -293,135 +320,136 @@ const Api = {
+                                            /**
+                                                grouping all tags with same property together (i.e <dc:subject>mobilità</dc:subject> <dc:subject>traffico</dc:subject>)
+                                                will become { subject: ["mobilità", "traffico"] }
+                                            **/
+                                            if (dc[elName] && Array.isArray(dc[elName])) {
+                                                dc[elName].push(finalEl);
+                                            } else if (dc[elName]) {
+                                                dc[elName] = [dc[elName], finalEl];
                                             } else {
-                                                _dcRef = _dcRef.concat(URIs);
+                                                dc[elName] = finalEl;
                                             }
-                                            obj.dc = dc;
                                         }
-                                        records.push(obj);
+                                        const URIs = castArray(dc.references.length > 0 ? dc.references : dc.URI);
+                                        if (!_dcRef) {
+                                            _dcRef = URIs;
+                                        } else {
+                                            _dcRef = _dcRef.concat(URIs);
+                                        }
+                                        obj.dc = dc;
                                     }
+                                    records.push(obj);
                                 }
-                                result.records = records;
-                                return addCapabilitiesToRecords(_dcRef, result, options);
-                            } else if (json && json.name && json.name.localPart === "ExceptionReport") {
-                                return {
-                                    error: json.value.exception && json.value.exception.length && json.value.exception[0].exceptionText || 'GenericError'
-                                };
                             }
+                            result.records = records;
+                            return addCapabilitiesToRecords(_dcRef, result, options);
+                        } else if (json && json.name && json.name.localPart === "ExceptionReport") {
+                            return {
+                                error: json.value.exception && json.value.exception.length && json.value.exception[0].exceptionText || 'GenericError'
+                            };
                         }
-                        return null;
-                    }));
+                    }
+                    return null;
+                }).then(results=>getBboxFor3DLayersToRecords(results)));         // handle getting bbox from capabilities in case of 3D tile layer
```

Comment:
```suggestion
                }).then(results => getBboxFor3DLayersToRecords(results)));         // handle getting bbox from capabilities in case of 3D tile layer
```